### PR TITLE
Deletion

### PR DESCRIPTION
**Pull request by meinou**

### CODE REVIEW

File: apps/blog/src/app/app.component.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 
-<div><h2><span>Angular Codelab Newsletter</span></h2><img src="/assets/images/logo.png"/></div>
+<div [routerLink]="['']"><h2><span>Angular Codelab Newsletter</span></h2><img src="/assets/images/logo.png"/></div>
```
Comment (kirjs) [inline apps/blog/src/app/app.component.html]: Would `routerLink="/"` work

File: apps/blog/src/app/form.service.ts [hunk 2]
```diff
@@ -7,7 +7,10 @@ import { Post } from './form/form.component';
   providedIn: 'root'
 })
 export class FormService {
-  repo$: AngularFireList<Post> = this.database.list('/posts');
+  all$: AngularFireList<Post> = this.database.list('/posts');
```
Comment (kirjs) [inline apps/blog/src/app/form.service.ts]: Do you need all$?

File: apps/blog/src/app/form/form.component.ts [hunk 3]
```diff
@@ -43,8 +46,11 @@ export class FormComponent {
       .addPost(
         formValues
       )
-      .then(() => {
+      .then((result) => {
         this.myform.reset();
+        console.log(result);
```
Comment (kirjs) [inline apps/blog/src/app/form/form.component.ts]: remove

File: apps/blog/src/app/form/form.component.ts [hunk 4]
```diff
@@ -43,8 +46,11 @@ export class FormComponent {
...
+        const key = result.key;
```
Comment (kirjs) [inline apps/blog/src/app/form/form.component.ts]: Inline this?

File: apps/blog/src/app/post.service.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+})
+export class PostService {
+    repo$: AngularFireList<Post> = this.database.list('/posts');
+
+    constructor(
+        private database: AngularFireDatabase) {
+    }
+
+    removePost(id: string) {
+        // console.log(id);
```
Comment (kirjs) [inline apps/blog/src/app/post.service.ts]: remove

File: apps/blog/src/app/form.service.ts [hunk 6]
```diff
@@ -7,7 +7,10 @@ import { Post } from './form/form.component';
...
+  repo$: AngularFireList<Post> = this.database.list('/posts', (ref) => {
```
Comment (kirjs) [inline apps/blog/src/app/form.service.ts]: Why is this form service thing is needed at all? Just use post.service for everything?

File: apps/blog/src/app/post.service.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        return this.database.object(`posts/${id}`).remove()
+            .then(result => {
+                console.log('done with ', result);
```
Comment (kirjs) [inline apps/blog/src/app/post.service.ts]: Drop this.

File: apps/blog/src/app/single-post/single-post.component.html [hunk 8]
```diff
@@ -11,6 +11,8 @@
       </markdown>
 
       <span *ngIf="!full"><a routerLink="/post/{{post.key}}">More</a> </span>
+      <button *ngIf="full" (click)="edit()">Edit</button>
```
Comment (kirjs) [inline apps/blog/src/app/single-post/single-post.component.html]: Block edit and delete in one ngIf block

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 9]
```diff
@@ -13,4 +16,27 @@ import { ActivatedRoute } from '@angular/router';
...
+ @Input() key = '';
+
+  constructor
+  (
+    private postService: PostService,
+    private accessService: AccessService,
+              private router: Router) {  }
+
+ delete() {
+    this.accessService.oldIsAdmin$.subscribe(console.log);
```
Comment (kirjs) [inline apps/blog/src/app/single-post/single-post.component.ts]: Remove console .og

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 10]
```diff
@@ -13,4 +16,27 @@ import { ActivatedRoute } from '@angular/router';
...
+    console.log('id: ', this.key);
+    this.post.hidden = true;
+    this.postService
+      .updatePost(this.key, this.post)
+      .then((result) => {
+        // debugger;
```
Comment (kirjs) [inline apps/blog/src/app/single-post/single-post.component.ts]: Remove

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 11]
```diff
@@ -13,4 +16,27 @@ import { ActivatedRoute } from '@angular/router';
...
+        // console.log('result: ', result);
+        this.router.navigateByUrl(``);
```
Comment (kirjs) [inline apps/blog/src/app/single-post/single-post.component.ts]: /

File: apps/blog/src/app/form/form.component.ts [hunk 12]
```diff
@@ -2,14 +2,16 @@ import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable } from 'rxjs';
 import { FormControl, FormGroup, Validators } from '@angular/forms';
-import { FormService } from '../form.service';
+import { PostService } from '../post.service';
+import { Router } from '@angular/router';
 
 export interface Post {
```
Comment (NothingEverHappens) [inline apps/blog/src/app/form/form.component.ts]: This should live in some common place, e.g. `common.ts`

File: apps/blog/src/app/post.service.ts [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+    repo$: AngularFireList<Post> = this.database.list('/posts', (ref) => {
+        return ref.orderByChild('hidden').equalTo(null);
+      });
+
+    constructor(
+        private database: AngularFireDatabase) {
+    }
+
+    removePost(id: string) {
+        return this.database.object(`posts/${id}`).remove();
```
Comment (NothingEverHappens) [inline apps/blog/src/app/post.service.ts]: can you abstract out 
 this.database.object(`posts/${id}`)?
As it is used 3 times

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 14]
```diff
@@ -13,4 +15,25 @@ import { ActivatedRoute } from '@angular/router';
...
+    private accessService: AccessService,
+    private router: Router,
+    private snackBar: MatSnackBar) {  }
+
+ delete() {
+    this.accessService.oldIsAdmin$.subscribe();
+    this.post.hidden = true;
+    this.postService
+      .updatePost(this.key, this.post)
+      .then((result) => {
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.ts]: You don't need result here

File: apps/blog/src/app/single-post/single-post.component.ts [hunk 15]
```diff
@@ -13,4 +15,25 @@ import { ActivatedRoute } from '@angular/router';
...
+        this.router.navigateByUrl(``);
+      })
+      .catch((err) => {
+        this.snackBar.open(`ERR: ${err}`);
```
Comment (NothingEverHappens) [inline apps/blog/src/app/single-post/single-post.component.ts]: ERR -> Something human readable?

File: apps/codelab/src/app/modules/admin/services/can-activate-child.guard.ts [hunk 16]
```diff
@@ -1,6 +1,6 @@
 import { Injectable } from '@angular/core';
 import { CanActivateChild } from '@angular/router';
-import { AccessService, Permissions } from './access.service';
+import { AccessService, Permissions } from '../../../../../../../libs/firebase-login/src/lib/access.service';
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/modules/admin/services/can-activate-child.guard.ts]: I think you can just use here
import { AccessService } from '@codelab/firebase-login';

File: apps/codelab/src/app/modules/admin/users/users.component.ts [hunk 17]
```diff
@@ -4,7 +4,7 @@ import { SyncDbService } from '@codelab/utils/src/lib/sync/services/sync-db.serv
 import { map } from 'rxjs/operators';
 import { firebaseToValuesWithKey } from '@codelab/utils/src/lib/sync/common';
 import { combineLatest, Observable } from 'rxjs';
-import { Permissions } from '../services/access.service';
+import { Permissions } from '../../../../../../../libs/firebase-login/src/lib/access.service';
```
Comment (NothingEverHappens) [inline apps/codelab/src/app/modules/admin/users/users.component.ts]: You are not using permissions

File: apps/codelab/src/app/modules/admin/services/can-activate-child.guard.ts (hunk 16 above)
Comment (NothingEverHappens) [inline apps/codelab/src/app/modules/admin/services/can-activate-child.guard.ts]: You are not using permissions

File: libs/feedback/src/lib/feedback-widget/feedback-widget.component.ts [hunk 18]
```diff
@@ -10,7 +10,7 @@ import { Message } from '../message';
 import { Observable, Subject } from 'rxjs';
 import { FormBuilder, FormGroup, Validators } from '@angular/forms';
 import { debounceTime, takeUntil } from 'rxjs/operators';
-import { AccessService } from '../../../../../apps/codelab/src/app/modules/admin/services/access.service';
+import { AccessService } from '../../../../firebase-login/src/lib/access.service';
```
Comment (NothingEverHappens) [inline libs/feedback/src/lib/feedback-widget/feedback-widget.component.ts]: I think you can just use here
import { AccessService } from '@codelab/firebase-login';

File: apps/blog/src/app/form/form.component.ts [hunk 19]
```diff
@@ -34,17 +36,19 @@ export class FormComponent {
...
 
   onSubmit() {
     const formValues: any = this.myform.getRawValue();
-    this.formService
+    this.postService
       .addPost(
         formValues
       )
-      .then(() => {
+      .then((result) => {
```
Comment (NothingEverHappens) [inline apps/blog/src/app/form/form.component.ts]: nit: you can do
`then(({key}) => {`

 this.router.navigateByUrl(`post/${key}`);

File: apps/blog/src/app/single-post/single-post.component.ts (hunk 15 above)
Comment (meinou) [reply apps/blog/src/app/single-post/single-post.component.ts]: I prefer robots

File: apps/blog/src/app/post.service.ts [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+    updatePost(id: string, post: Partial<Post>) {
+        return this.getPostById(id).update(post);
+    }
+
+    addPost(
+        post: Post
+      ): any {
+        const newpost: Post = {
+          ...post,
+          date: new Date().toUTCString()
```
Comment (kirjs) [inline apps/blog/src/app/post.service.ts]: use this: 
database.ServerValue.TIMESTAMP

File: apps/blog/src/app/post/post.component.html [hunk 21]
```diff
@@ -1,3 +1,5 @@
 <ng-container *ngIf="post$ | async as post">
-  <codelab-single-post [post]=post [full]="true"></codelab-single-post>
+  <codelab-single-post [post]=post [full]="true" [key]="key"></codelab-single-post>
+  <button routerLink="">Back</button>
```
Comment (kirjs) [inline apps/blog/src/app/post/post.component.html]: /

**Comment by meinou** [comment]
@kirjs how I can resolve conflicts? (I did not change those files)

**Pull request merged**